File: 48-rotate-image/48-rotate-image.js
/**
 * @param {number[][]} matrix
 * @return {void} Do not return anything, modify matrix in-place instead.
 */
var rotate = function(matrix) {
    const n = matrix.length;
    const m = matrix[0].length;
    const origin = []
    matrix.map((mat) => origin.push([...mat]))

    for (let i=0; i<n; i++){
        for (let j=0; j<m; j++){
            matrix[j][n-i-1] = origin[i][j]
        }
    }
    
    
    return
};

File: 48-rotate-image/48-rotate-image.py
class Solution:
    def rotate(self, matrix: List[List[int]]) -> None:
        """
        Do not return anything, modify matrix in-place instead.
        """
        n = len(matrix)
        result = [[0] * n for _ in range(n)]
        
        for y in range(n) :
            for x in range(n) :
                result[x][n-1-y] = matrix[y][x]
        
        matrix[:] = result